### ISSUE TITLE
How to change ImGui!

### ISSUE THREAD

**Issue by ROCKIv** [danielkrupinski/Osiris#2918]
Hi! I want to learn how to change osiris imgui to for example blaster.vip gui. Dont hate me, i know a little about c++, if u can please dont tell me "Change calls" or somethings like that because i dont understand anything. If u want to help me tell me at discord: spec#9313

Thanks for your time, see you son!

**Comment by wemanzoz** [comment]
go to gui.cpp then do your imgui thing

**Comment by ROCKIv** [comment]
> go to gui.cpp then do your imgui thing

yes but i have a lot of errors, if u can help me i can screen share u :)  | spec#9313

**Comment by Arthurgroll997** [comment]
First things first, compile the project. After that, you can create a new header file and a cpp file modifying everything you need, or you can simply modify imgui.cpp and change the colors and shapes you are willing to change. Still, have in mind that you'll need to understand c++ to do so. I recommend you searching for imgui templates, after that just copy and paste the ones you like the most. (sorry for bad english)

**Comment by linuxer9** [comment]
don't waste your time changing the ui, the goal of any ui is to get your job done. unless if you are trying to sell it.
default Imgui is quite nice looking, but you can make it look even nicer by just changing colors and some styling.
just add this line `ImGui::ShowDemoWindow();` in GUI::render() function, this will show you a window where you can play with styling, colors, spacing ...etc. then copy that code and add it to `GUI::GUI()`.
ez pasta..

**Comment by ROCKIv** [comment]
Hi! @linuxer9 i paste ImGui::ShowDemoWindow(); in void GUI::render() noexcept without any problem. But i dont know the GUI::GUI() part, like i need to replace this 

    ImGui::StyleColorsDark();
    ImGuiStyle& style = ImGui::GetStyle();

    style.ScrollbarSize = 9.0f;

    ImGuiIO& io = ImGui::GetIO();
    io.IniFilename = nullptr;
    io.LogFilename = nullptr;
    io.ConfigFlags |= ImGuiConfigFlags_NoMouseCursorChange;

    ImFontConfig cfg;
    cfg.SizePixels = 15.0f;

#ifdef _WIN32
    if (PWSTR pathToFonts; SUCCEEDED(SHGetKnownFolderPath(FOLDERID_Fonts, 0, nullptr, &pathToFonts))) {
        const std::filesystem::path path{ pathToFonts };
        CoTaskMemFree(pathToFonts);

        fonts.normal15px = io.Fonts->AddFontFromFileTTF((path / "tahoma.ttf").string().c_str(), 15.0f, &cfg, Helpers::getFontGlyphRanges());
        if (!fonts.normal15px)
            io.Fonts->AddFontDefault(&cfg);

        cfg.MergeMode = true;
        static constexpr ImWchar symbol[]{
            0x2605, 0x2605, // ★
            0
        };
        io.Fonts->AddFontFromFileTTF((path / "seguisym.ttf").string().c_str(), 15.0f, &cfg, symbol);
        cfg.MergeMode = false;
    }
#else
    fonts.normal15px = addFontFromVFONT("csgo/panorama/fonts/notosans-regular.vfont", 15.0f, Helpers::getFontGlyphRanges(), false);
#endif
    if (!fonts.normal15px)
        io.Fonts->AddFontDefault(&cfg);
    addFontFromVFONT("csgo/panorama/fonts/notosanskr-regular.vfont", 15.0f, io.Fonts->GetGlyphRangesKorean(), true);
    addFontFromVFONT("csgo/panorama/fonts/notosanssc-regular.vfont", 17.0f, io.Fonts->GetGlyphRangesChineseFull(), true);


to my imgui template? like i have this template https://www.youtube.com/watch?v=0mqGGwT0HiE

**Comment by Toms0n** [comment]
Dont paste that is all, why u want change menu? U want sell this? If u dont know a little c++ then dont do p2c pls!
U have all documentation in imgui github, its so easy to understand.

**Comment by ROCKIv** [comment]
hi! i dont want to sell it, i want to learn how to change the imgui! Sadly i dont understand how to do it, for that im doing this topic :)

**Comment by VuBrian22** [comment]
> hi! i dont want to sell it, i want to learn how to change the imgui! Sadly i dont understand how to do it, for that im doing this topic :)

There are thousands of other projects out there that use ImGUI. Why don't you just use a much simpler application with ImGUI for experimentation especially since you don't know C++ well. It could be a basic calculator app made in C++ with ImGUI. I don't understand why you would need something advanced like a csgo cheat to "practice and learn" on. That just doesn't make sense to me. Unless ofc you are planning to sell the cheat under a new GUI and Brand. That is hella scummy btw.

**Comment by ROCKIv** [comment]
Hi brian! i want to learn how to code cheats, doing features like ragebot its so difficult! I Want to start doing things like chaning imgui, adding features from other cheats, etc!
